lib/pbench/agent/tool_meister.py

```diff
@@ -328,7 +328,12 @@ def __init__(self, name, tool_opts, logger=None, **kwargs):
         self.pmcd_process = None
         self.pmlogger_process = None
         if self.tool_dir:
-            self.tool_dir.mkdir()
+            try:
+                self.tool_dir.mkdir()
+            except Exception as exc:
+                self.logger.error(
+                    "Failed to create tool directory '%s': '%s'", self.tool_dir, exc
+                )
```

Comment:
Is there a scenario in which this directory would already exist, in which case we might not want to bomb out, here?  (E.g., if I create the tool, then remove it, then try to create it again?)  Also, this directory is/could be on the remote system where the benchmark workload will run (right?), so it might be inconvenient to delete this directory once it has been created.

This is all a long-winded way of asking whether we should be specifying `exists_ok=True` on the `mkdir()` call....

Reply:
I don't think a scenario currently exists, as other samples/iterations will be under different parent directories, and the whole run will be under a timestamped directory, so no room for re-use currently. Also no way for it to (intentionally at least) attempt to create multiple times for a single sample/iteration. There could potentially be optimizations in the future that involve reusing the created dir though, so perhaps useful for the future.